### ISSUE TITLE
Loss function value is nan 

### ISSUE THREAD

**Issue by Eloneinstein** [sgrvinod/a-PyTorch-Tutorial-to-Object-Detection#53]
The multi box loss value is showing more than 2000 initially and loss value is nan. Why is it so? @sgrvinod

**Comment by toomy0toons** [comment]
try reducing batch size and gradient clipping as sgrvinod noted

im using batch size 8 and gradient clipping value 1

**Comment by Eloneinstein** [comment]
Thanks for the response. I found that  log of negative decimal in the cxcy_to_gcxgcy function gives nan values.. I have to make them absolute to gives proper values. How comes you did not get any nan values?. If the input to the log function is negative decimal it gives nan.

**Comment by toomy0toons** [comment]
@Eloneinstein 
center cords should be positive as it measures distance in ratio so the function should always return positive. 
are you using a custom dataset? 
if so match the input to xmin,ymin,xmax,ymax. Same error happened to me when i was feeding different dataset which is was xmax,ymin, xmin, ymax

**Comment by Eloneinstein** [comment]
@toomy0toons .  I am using the same VOC 2007 dataset. I will check for the order of the coordinates.

**Comment by Eloneinstein** [comment]
@toomy0toons . Thanks a lot for the solution. I put the coordinates in the wrong order. Thanks again.
